administrator/components/com_templates/Model/TemplateModel.php

```diff
@@ -148,6 +148,151 @@ public function getDirectoryTree($dir)
+		if (stristr($type, 'mod_') != false)
+		{
+			$folder   = $explodeArray['2'];
+			$htmlPath = \JPath::clean($modulePath . $folder . '/tmpl/');
+			$fileName = $this->getSafeName($fileName);
+			$coreFile = $this->findPath($htmlPath, $fileName);
+
+			return $coreFile;
+		}
+		elseif (stristr($type, 'com_') != false)
+		{
+			// For components
+			$folder    = $explodeArray['2'];
+			$subFolder = $explodeArray['3'];
+			$fileName  = $this->getSafeName($fileName);
+
+			// The old scheme, if a view has a tmpl folder
+			$oldHtmlPath = \JPath::clean($componentPath . $folder . '/views/' . $subFolder . '/tmpl/');
+
+			if(!$coreFile = \JPath::find($oldHtmlPath, $fileName))
+			{
+				// The new scheme, the views are directly in the component/tmpl folder
+				$newHtmlPath = \JPath::clean($componentPath . $folder . '/tmpl/' . $subFolder . '/');
+				$coreFile    = $this->findPath($newHtmlPath, $fileName);
+
+				return $coreFile;
+			}
+
+			return $coreFile;
+		}
+		elseif (stristr($type, 'layouts') != false)
+		{
+			// For Jlayouts
+			$subtype = $explodeArray['3'];
+
+			if(stristr($subtype, 'com_'))
+			{
+				$folder    = $explodeArray['3'];
+		 		$subFolder = array_slice($explodeArray, 4, -1);
+				$subFolder = implode(DIRECTORY_SEPARATOR, $subFolder);
+				$htmlPath  = \JPath::clean($componentPath . $folder . '/layouts/' . $subFolder);
+				$fileName  = $this->getSafeName($fileName);
+				$coreFile  = $this->findPath($htmlPath, $fileName);
+
+				return $coreFile;
+			}
+			elseif (stristr($subtype, 'joomla') || stristr($subtype, 'libraries') || stristr($subtype, 'plugins'))
+			{
+				$subFolder = array_slice($explodeArray, 3, -1);
+				$subFolder = implode(DIRECTORY_SEPARATOR, $subFolder);
+				$htmlPath  = \JPath::clean($layoutPath . $subFolder);
+				$fileName  = $this->getSafeName($fileName);
+				$coreFile  = $this->findPath($htmlPath, $fileName);
+
+				return $coreFile;
+			}
+		}
+
+		$app->enqueueMessage(\JText::_('COM_TEMPLATES_ERROR_CORE_FILE_NOT_FOUND'), 'error');
+
+		return false;
+	}
+
+	/**
+	 * Creates a safe file name for the given name.
+	 *
+	 * @param   string  $name  The filename
+	 *
+	 * @return  string $fileName  The filtered name without Date
+	 *
+	 * @since   4.0.0
+	 */
+	private function getSafeName($name)
+	{
+		if (preg_match('/[0-9]/', $name))
+		{
+			// Get the extension
+			$extension = \JFile::getExt($name);
+
+			// Remove ( Date ) from file
+			$explodeArray         = explode('-', $name);
+			$nameWithoutExtension = implode('-', array_slice($explodeArray, 0, -2));
+
+			//Filtered name
+			$name = $nameWithoutExtension . '.' . $extension;
+		}
+
+		return $name;
+	}
+
+	/**
+	 * Find file in given folde
+	 *
+	 * @param  string  $path An path or array of path to search.
+	 * @param  string  $file The file name to look for.
+	 *
+	 * @return mixed  $path The full path and file name for the target file, or boolean false if the file is not found in any of the paths.
+	 *
+	 * @since  4.0.0
```

Comment:
Since tags should be `__DEPLOY_VERSION__` we actually do not know in which version this feature will land at the end.

Reply:
Ok